docs/src/use/migrate-to-9.0.0.md

```diff
@@ -465,6 +466,19 @@ ESLint v9.0.0 removes the deprecated `CodePath#currentSegments` property.
 
 **Related Issues(s):** [#17457](https://github.com/eslint/eslint/issues/17457)
 
+## <a name="codepath-precalc"></a> Code paths are now precalculated
+
+Prior to ESLint v9.0.0, code paths were calculated during the same AST traversal used by rules, meaning that the information passed to methods like `onCodePathStart` and `onCodePathSegmentStart` was incomplete. Specifically, array properties like `CodePath#childCodePaths` and `CodePathSegment#nextSegments` began empty and then were filled with the appropriate information as the traversal completed, meaning that those arrays could have different elements depending on when you checked their values.
+
+ESLint v9.0.0 now precalculates code path information before the traversal used by rules. As a result, the code path information is now complete regardless of where it is accessed inside of a rule.
+
+**To address:** If you are accessing any array properties on `CodePath` or `CodePathSegment`, you'll need to update your code. Specifically:
+
+* If you are checking the `length` of any array properties, ensure you are using relative comparison operators like `<`, `>`, `<=`, and `>=` instead of equals.
```

Comment:
This is one of the bugs I ran into when implementing this, where we were doing `=== 0` in one spot. I think it's worth calling this out because it took me a while to realize why that was a problem.